src/main/resources/static/js/popup/ConnectBase.js

```diff
@@ -0,0 +1,23 @@
+class ConnectBase {
+  constructor() {
+    this._url = "http://localhost:3000/auth/";
+    this.headers = {
+      "Content-Type": "application/json"
+    };
+  }
+
+  async postRequest(body, value) {
+    let url;
+    if (value.classList[1] === "btn-login") {
+      url = this._url + "login";
+    } else {
+      url = this._url + "register";
+      document.querySelector(".auth").style.display = "none";
+    }
+    return await fetch(url, {
+      method: "POST",
+      headers: this.headers,
+      body: JSON.stringify(body)
+    }).then(data => data.json());
```

Comment:
You should add exception handling also or make few attempts for sending the request. We have pretty dumb server so I couldn't guarantee that your request will be send from first attempt.
Also from back side we response with different codes and bodies. Try it with swagger-ui: http://54.37.125.180:8080/swagger-ui.html#/